src/overlays/actors/ovl_Bg_Dy_Yoseizo/z_bg_dy_yoseizo.c

```diff
@@ -71,55 +64,978 @@ const ActorInit Bg_Dy_Yoseizo_InitVars = {
+                particleInitPos.x = this->actor.posRot.pos.x;
+                particleInitPos.y = this->actor.posRot.pos.y + spawnPosVariation +
+                                    ((Rand_ZeroOne() - 0.5f) * (spawnPosVariation * 0.5f));
+                particleInitPos.z = this->actor.posRot.pos.z + 30.0f;
+            } else {
+                particleLife = 50;
+                particleType = type;
+                particleScale = 0.2f;
+                particleInitPos.x = this->actor.posRot.pos.x + Rand_CenteredFloat(10.0f);
+
+                if (globalCtx->sceneNum == SCENE_DAIYOUSEI_IZUMI) {
+                    particleInitPos.y = this->actor.posRot.pos.y + spawnPosVariation + 50.0f +
+                                        ((Rand_ZeroOne() - 0.5f) * (spawnPosVariation * 0.1f));
+                    particleInitPos.z = this->actor.posRot.pos.z + 30.0f;
+                } else {
+                    particleInitPos.y = this->actor.posRot.pos.y + spawnPosVariation - 30.0f +
+                                        ((Rand_ZeroOne() - 0.5f) * (spawnPosVariation * 0.1f));
+                    particleInitPos.z = this->actor.posRot.pos.z + 60.0f;
+                }
+
+                if (LINK_IS_ADULT) {
+                    particleInitPos.y += 20.0f;
+                }
+            }
+
+            particlePrimColor.r = sParticlePrimColors[particleType].r;
+            particlePrimColor.g = sParticlePrimColors[particleType].g;
+            particlePrimColor.b = sParticlePrimColors[particleType].b;
+            particleEnvColor.r = sParticleEnvColors[particleType].r;
+            particleEnvColor.g = sParticleEnvColors[particleType].g;
+            particleEnvColor.b = sParticleEnvColors[particleType].b;
+            BgDyYoseizo_ParticleInit(this, &particleInitPos, &particleInitVelocity, &particleAccel, &particlePrimColor,
+                                     &particleEnvColor, particleScale, particleLife, particleType);
+        }
+    }
+}
+
+void BgDyYoseizo_Bob(BgDyYoseizo* this, GlobalContext* globalCtx) {
+    this->targetHeight = this->grownHeight + this->bobOffset;
+    Math_ApproachF(&this->actor.posRot.pos.y, this->targetHeight, 0.1f, 10.0f);
+    Math_ApproachF(&this->bobOffset, 10.0f, 0.1f, 0.5f);
+
+    if (globalCtx->csCtx.state == 0) {
+        this->actor.velocity.y = Math_SinS(this->bobTimer);
+    } else {
+        this->actor.velocity.y = Math_SinS(this->bobTimer) * 0.4f;
+    }
+}
+
+void BgDyYoseizo_CheckMagicAcquired(BgDyYoseizo* this, GlobalContext* globalCtx) {
+    if (Flags_GetSwitch(globalCtx, 0x38)) {
+        globalCtx->msgCtx.unk_E3EE = 4;
+        if (globalCtx->sceneNum == SCENE_DAIYOUSEI_IZUMI) {
+            if (!(gSaveContext.magicAcquired) && (this->fountainType != FAIRY_UPGRADE_SPIN_ATTACK)) {
+                Actor_Kill(&this->actor);
+                return;
+            }
+        } else if (!(gSaveContext.magicAcquired)) {
+            Actor_Kill(&this->actor);
+            return;
+        }
```

Comment:
```suggestion
        } else {
            if (!(gSaveContext.magicAcquired)) {
                Actor_Kill(&this->actor);
                return;
            }
        }
```
This has better parallelism with the previous clause.